app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -536,12 +536,18 @@ class PatientRepository @Inject constructor(
   }
 
   fun bpPassportForPatient(patientUuid: UUID): Single<Optional<BusinessId>> {
-    return Single.fromCallable {
-      database
-          .businessIdDao()
-          .latestIdForType(patientUuid, BpPassport)
-          .toOptional()
-    }
+    return database
+        .businessIdDao()
+        .latestIdForType(patientUuid, BpPassport)
+        .toObservable()
```

Comment:
Can you move this `.toObservable()` to after the `.map { }` operator?

Reply:
In fact, if you still need to publish it as a `Single`, you don't need a `.toObservable()` here, because `Flowable` also has a `firstOrError()` operator that converts it to a `Single`.